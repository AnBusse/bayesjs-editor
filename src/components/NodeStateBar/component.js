import React, { Fragment } from 'react';

import PropTypes from 'prop-types';

const NodeStateBar = ({
  barWidth,
  fillColor,
  percentText,
  x,
  textX,
  textY,
  barX,
  barY,
  height,
  width,
  onDoubleClick,
}) => (
  <Fragment>
    <rect
      x={barX}
      y={barY}
      height={height}
      width={barWidth}
      fill={fillColor}
    />

    <text
      x={textX}
      y={textY}
      height={height}
      fontSize="14px"
      alignmentBaseline="rigth"
      stroke="black"
      strokeWidth="1.1"
    >
      {percentText}
    </text>

    <rect
      x={x}
      y={barY}
      height={height}
      width={width}
      fill="transparent"
      stroke="#333"
      strokeWidth="1"
      onDoubleClick={onDoubleClick}
    >
      <title>
        {percentText}
      </title>
    </rect>
  </Fragment>
);

NodeStateBar.propTypes = {
  barWidth: PropTypes.number.isRequired,
  fillColor: PropTypes.string.isRequired,
  percentText: PropTypes.string.isRequired,
  textX: PropTypes.number.isRequired,
  textY: PropTypes.number.isRequired,
  barX: PropTypes.number.isRequired,
  barY: PropTypes.number.isRequired,
  x: PropTypes.number.isRequired,
  height: PropTypes.number.isRequired,
  width: PropTypes.number.isRequired,
  onDoubleClick: PropTypes.func.isRequired,
};

export default NodeStateBar;
