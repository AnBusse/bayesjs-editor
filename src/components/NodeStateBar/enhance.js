import { withProps } from 'recompose';

const getResult = (results, state, belief) => {
  if (belief !== null) {
    return belief === state ? 1 : 0;
  }

  return results[state];
};

const getTextX = (x, width, result) => {
  const finalX = x + (width / 2) - 18;

  return result === 1 ? finalX - 4 : finalX;
};

const enhance = withProps(({
  results,
  state,
  belief,
  x,
  y,
  width,
}) => {
  const result = getResult(results, state, belief);

  return {
    textX: getTextX(x, width, result),
    textY: y + 12,
    barX: x + (width - (width * result)),
    barY: y,
    barWidth: width * result,
    percentText: `${(100 * result).toFixed(2)} %`,
    fillColor: belief != null ? '#EE4040' : '#9f9ff6',
  };
});

export default enhance;
