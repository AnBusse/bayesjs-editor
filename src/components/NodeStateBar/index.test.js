import React from 'react';
import { shallow } from 'enzyme';
import NodeStateBar from './component';

const defaultProps = {
  barWidth: 70,
  fillColor: '#ddd',
  percentText: '50 %',
  x: 5,
  textX: 5,
  textY: 0,
  barX: 80,
  barY: 5,
  height: 20,
  width: 70,
  onDoubleClick: jest.fn(),
};

const shallowComponent = (props = {}) => {
  const compProps = { ...defaultProps, ...props };

  return shallow(<NodeStateBar {...compProps} />);
};

describe('NodeStateBar Component', () => {
  it('matches snapshot', () => {
    const component = shallowComponent();

    expect(component).toMatchSnapshot();
  });

  describe('Event Listeners', () => {
    describe('onStateDoubleClick', () => {
      const onDoubleClick = jest.fn();
      const component = shallowComponent({ onDoubleClick });

      beforeAll(() => {
        component.find('rect').last().simulate('doubleClick');
      });

      it('triggers "onDoubleClick" on doubleClick', () => {
        expect(onDoubleClick).toBeCalled();
      });
    });
  });
});
