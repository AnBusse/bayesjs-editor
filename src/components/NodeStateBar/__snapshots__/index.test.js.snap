// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NodeStateBar Component matches snapshot 1`] = `
<Fragment>
  <rect
    fill="#ddd"
    height={20}
    width={70}
    x={80}
    y={5}
  />
  <text
    alignmentBaseline="rigth"
    fontSize="14px"
    height={20}
    stroke="black"
    strokeWidth="1.1"
    x={5}
    y={0}
  >
    50 %
  </text>
  <rect
    fill="transparent"
    height={20}
    onDoubleClick={[Function]}
    stroke="#333"
    strokeWidth="1"
    width={70}
    x={5}
    y={5}
  >
    <title>
      50 %
    </title>
  </rect>
</Fragment>
`;
